### ISSUE TITLE
macOS - drag a feed from iCloud list to a folder in iCloud list in another NNW window fails

### ISSUE THREAD

**Issue by haikusw** [Ranchero-Software/NetNewsWire#3580]
Context: Syncing with iCloud, macOS NNW Version 6.1 (6106), macOS 12.3.1 (21E258), 13" M1 MBP.

1. Open two NNW windows
2. drag a feed item from the top level of the iCloud group to a folder in the iCloud group in the second window.
-> Drag seems like it will work but the feed item just springs back.

**Comment by brentsimmons** [comment]
Does the same move work when you’re dragging within the same window?

**Comment by haikusw** [comment]
Yes.  I've downloaded the source and am taking a look now.  Will submit a PR if I can replicate in Xcode and see an obvious fix.

**Comment by haikusw** [comment]
Issue seems to be here:

https://github.com/Ranchero-Software/NetNewsWire/blob/b52292d4590ab07616981dffd5956aec6deea15f/Mac/MainWindow/Sidebar/SidebarOutlineDataSource.swift#L373-L376

code assumes `draggedNodes` is set but it's set in the _source_ outline view, not in the destination outline view and so when the outline view in a different window is accepting the drop it returns false here because `draggedNodes` is unset.

**Comment by haikusw** [comment]
(digging further…)

**Comment by haikusw** [comment]
(naming! maybe `localDragFeedPasteboardOperation` instead?)

**Comment by haikusw** [comment]
Next step (for feeds) would be to address `draggedNodes` assumption in: `acceptLocalFeedsDrop(…)`.

I'm find pursuing this path if you think it sounds reasonable.  Just didn't want to do a lot of work down a path that isn't how you'd like this fixed.

(It would be handy if `PasteboardWebFeed` could be used to initialize a `WebFeed`; just ran into the second time that would have allowed for less duplication of code.)

Thanks.

**Comment by haikusw** [comment]
Back to this after a long break.
Got the most basic version working: local feed single and multiple item copy & move between windows within same account.  Some of the code is a bit more complicated than I'd like, but `PasteboardWebFeeds` have pretty limited information.

Still to do:
- copy between accounts
- fix folder drags between accounts works across windows
- Step back and see if I can simplify the approach based on what I've learned about how things work (switch from nodes to feeds earlier?).

**Comment by haikusw** [comment]
@brentsimmons draft PR with the fix is up.  As noted in the PR, I'd like to remove `draggedNodes` entirely and reduce the complexity and code thereby considerably but wanted to check with you first.  Let me know.

**Comment by haikusw** [comment]
Ugh. subtle bug to investigate.  sorry.
